ovirt/data_source_ovirt_clusters_test.go

```diff
@@ -4,63 +4,60 @@
 
-package ovirt
+package ovirt_test
 
 import (
+	"fmt"
 	"regexp"
 	"testing"
 
 	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
 )
 
 func TestAccOvirtClustersDataSource_nameRegexFilter(t *testing.T) {
+	suite := getOvirtTestSuite(t)
 	resource.Test(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
-		Providers: testAccProviders,
+		PreCheck:  suite.PreCheck,
+		Providers: suite.Providers(),
 		Steps: []resource.TestStep{
 			{
-				Config: testAccCheckOvirtClustersDataSourceNameRegexConfig,
+				Config: fmt.Sprintf(`
+data "ovirt_clusters" "name_regex_filtered_cluster" {
+  name_regex = "^%s$"
+}
+`, suite.Cluster().MustName()),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckOvirtDataSourceID("data.ovirt_clusters.name_regex_filtered_cluster"),
-					resource.TestCheckResourceAttr("data.ovirt_clusters.name_regex_filtered_cluster", "clusters.#", "2"),
-					resource.TestMatchResourceAttr("data.ovirt_clusters.name_regex_filtered_cluster", "clusters.0.name", regexp.MustCompile("^Default*")),
-					resource.TestMatchResourceAttr("data.ovirt_clusters.name_regex_filtered_cluster", "clusters.1.name", regexp.MustCompile("^Default*")),
+					suite.TestDataSource("data.ovirt_clusters.name_regex_filtered_cluster"),
+					resource.TestMatchResourceAttr("data.ovirt_clusters.name_regex_filtered_cluster", "clusters.0.name", regexp.MustCompile(fmt.Sprintf("^%s$", suite.Cluster().MustName()))),
 				),
 			},
 		},
 	})
 }
 
 func TestAccOvirtClustersDataSource_searchFilter(t *testing.T) {
+	suite := getOvirtTestSuite(t)
+
 	resource.Test(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
-		Providers: testAccProviders,
+		PreCheck:  suite.PreCheck,
+		Providers: suite.Providers(),
 		Steps: []resource.TestStep{
 			{
-				Config: testAccCheckOvirtClustersDataSourceSearchConfig,
+				Config: fmt.Sprintf(`
+data "ovirt_clusters" "search_filtered_cluster" {
+  search = {
+    criteria       = "name = %s"
+    max            = 1
+    case_sensitive = false
+  }
+}
+`, suite.Cluster().MustName()),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckOvirtDataSourceID("data.ovirt_clusters.search_filtered_cluster"),
 					resource.TestCheckResourceAttr("data.ovirt_clusters.search_filtered_cluster", "clusters.#", "1"),
-					resource.TestCheckResourceAttr("data.ovirt_clusters.search_filtered_cluster", "clusters.0.name", "Default"),
+					resource.TestCheckResourceAttr("data.ovirt_clusters.search_filtered_cluster", "clusters.0.name", suite.Cluster().MustName()),
 				),
 			},
 		},
 	})
-
-}
-
-var testAccCheckOvirtClustersDataSourceNameRegexConfig = `
-data "ovirt_clusters" "name_regex_filtered_cluster" {
-  name_regex = "^Default*"
-}
-`
-
-var testAccCheckOvirtClustersDataSourceSearchConfig = `
-data "ovirt_clusters" "search_filtered_cluster" {
-  search = {
-    criteria       = "name = Default"
-    max            = 1
-    case_sensitive = false
-  }
```

Comment:
Why did you remove the vars? I think it is much more comfortable

Reply:
In most places they needed an `Sprintf` to work with the dynamic updates. Having the string and the variables in different places made it difficult to sync. Later on I added a tool for using Go templates, but that's not 100% done everywhere.